app/bundles/EmailBundle/EventListener/PointSubscriber.php

```diff
@@ -115,12 +111,28 @@ public function onEmailOpen(EmailOpenEvent $event)
      */
     public function onEmailSend(EmailSendEvent $event)
     {
-        if ($leadArray = $event->getLead()) {
+        $leadArray = $event->getLead();
+        if ($leadArray && is_array($leadArray) && !empty($leadArray['id'])) {
             $lead = $this->entityManager->getReference(Lead::class, $leadArray['id']);
         } else {
             return;
         }
 
-        $this->pointModel->triggerAction('email.send', $event->getEmail(), null, $lead);
+        if ($this->shouldTriggerPointEmailSendAction($event, $lead)) {
```

Comment:
@kuzmany @dennisameling the issue of duplicated points was a bit higher. We called the event twice per email send. I discussed with @alanhartless that it was a bug. This commit fixes it so this check is not needed anymore.

https://github.com/mautic/mautic/pull/7716/commits/05bd59fa22494f861dc996e85f7a7ee7e83a89e0

Reply:
Then this part we can remove, right?

Reply:
Yes. Maybe I can remove it as part of https://github.com/mautic/mautic/pull/7716 that is fixing the double event dispatch if you agree.

Reply:
Yes, I've already seen it. Please do it as part of #7716 , then I'll make test.

Reply:
For the record, the code was removed as part of #7716